containers/cpp/.devcontainer/Dockerfile

```diff
@@ -2,6 +2,45 @@
 ARG VARIANT=debian-11
 FROM mcr.microsoft.com/vscode/devcontainers/cpp:0-${VARIANT}
 
+# Enable tab completion for bash and zsh
+# See https://github.com/microsoft/vcpkg#tab-completionauto-completion for more shell completion options
+ARG VCPKG_FORCE_SYSTEM_BINARIES=1
+RUN . /etc/os-release \
+    && if [ "$(dpkg --print-architecture)" = "amd64" ] || [ "${VERSION_CODENAME}" = "bullseye" ] || [ "${VERSION_CODENAME}" = "hirsute" ]; then \
+        su vscode -c "${VCPKG_ROOT}/vcpkg integrate bash" \
+        && su vscode -c "${VCPKG_ROOT}/vcpkg integrate zsh"; \
+    fi 
+
+# [Optional] Install CMake version different from base image.
+# [Choice] CMake Version: none, 3.21.5, 3.22.2, or versions from https://cmake.org/download/
+ARG REINSTALL_CMAKE_FROM_SOURCE="false"
+ARG CMAKE_VERSION="none"
+
+# Optionally install the cmake for vcpkg
+RUN if [ "${REINSTALL_CMAKE_FROM_SOURCE}" = "true" ] && [ "${CMAKE_VERSION}" != "none"  ]; then \
```

Comment:
qq: Is it possible to make this a script? Seems like there is substantial amount of work here that might be easier to maintain/enhance.

Reply:
yes, thx, done